satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -455,10 +452,40 @@ def is_high_resol(resolution):
 class DatasetWrapper:
     """Helper class for accessing the dataset."""
 
-    def __init__(self, nc):
+    def __init__(self, nc, decode_nc=True):
         """Wrap the given dataset."""
         self.nc = nc
 
+        if decode_nc is True:
+            self._decode_cf()
+            self._fix_duplicate_dimensions(self.nc)
+            self.nc = self._chunk(self.nc)
+
+    def _decode_cf(self):
+        # remove time before decoding and add again.
+        time = self.get_time()
+        time_dims = self.nc[time.name].dims
+        time = xr.where(time == time.attrs["_FillValue"], np.datetime64("NaT"),
+                        (time + time.attrs["add_offset"]).astype("datetime64[s]").astype("datetime64[ns]"))
+        self.nc = self.nc.drop_vars(time.name)
+        self.nc = xr.decode_cf(self.nc)
+        self.nc[time.name] = (time_dims, time.values)
+
+    def _fix_duplicate_dimensions(self, nc):
+        nc.variables["covariance_spectral_response_function_vis"].dims = ("srf_size_1", "srf_size_2")
+        self.nc = nc.drop_dims("srf_size")
+
+    def _chunk(self, nc):
+
+        (chunk_size_y, chunk_size_x) = nc.variables["quality_pixel_bitmask"].encoding["chunksizes"]
```

Comment:
Can you please add a comment here, that accessing the encoding doesn't load the data into memory? (Martin just raised the concern that this loads the bitmask array into memory. We tested and it doesn't)

Reply:
resolved because the function does not exist anymore